keras/utils/io_utils.py

```diff
@@ -83,7 +83,7 @@ def __getitem__(self, key):
                 idx = (self.start + key).tolist()
             else:
                 raise IndexError
-        elif isinstance(key, list):
+        elif isinstance(key, (list, six.moves.range)):
```

Comment:
Seems a bit specific. Since this is the last `if` clause, it would be okay to cast it to list:
```
else:
   key = list(key)
```

Reply:
Like so?
```
else: 
    key=list(key)
    if max(key) + self.start < self.end:
           idx = [x + self.start for x in key]
    else:
        raise IndexError
```

Reply:
Or even simpler:

```python
else:
    # Assume list/iterable
    if max(key) + self.start < self.end:
           idx = [x + self.start for x in key]
    else:
          raise IndexError
```